docs/src/main/asciidoc/redis-reference.adoc

```diff
@@ -0,0 +1,106 @@
+In order to do so, you will need:
+
+- Generate a new `Command` based on the nodejs code available in https://github.com/vert-x3/vertx-redis-client[`vertx-redis-client`] repository:
+
+If you don't have a redis service running locally, you can run redis in a docker container:
+[source,shell script]
+----
+docker run --name redis -p 7006:6379 -d redis
+----
+
+Next, while being in https://github.com/vert-x3/vertx-redis-client[`vertx-redis-client`] root folder execute:
+
+[source,shell script]
+----
+cd tools
+npm i
+npm start
+----
+
+The above sequence of commands should update the https://github.com/vert-x3/vertx-redis-client/blob/master/src/main/java/io/vertx/redis/client/Command.java[`Command.java`] file, so it includes all the possible commands supported by a particular redis version.
+
+[source,java]
+----
+Command ZUNION = Command.create("zunion", -3, 0, 0, 0, false, true, true, false);
+----
+
+This definition is very important as we will have to use it in the service.
+Once we have this Command we can start to update the redis-client extension by:
+
+- Updating the https://github.com/quarkusio/quarkus/blob/main/extensions/redis-client/runtime/src/main/java/io/quarkus/redis/client/RedisClient.java[`RedisClient`] interface, i.e.:
+
+[source,java]
+----
+Response zunion(List<String> args);
+----
+
+- Updating the https://github.com/quarkusio/quarkus/blob/main/extensions/redis-client/runtime/src/main/java/io/quarkus/redis/client/runtime/RedisClientImpl.java[`RedisClientImpl`], i.e.:
+
+[source,java]
+----
+@Override
+public Response zunion(List<String> args) {
+    final io.vertx.mutiny.redis.client.Command ZUNION = Command.create("zunion", -3, 0, 0, 0, false, true, true, false);
+    final io.vertx.mutiny.redis.client.Request requestWithArgs = args.stream().reduce(
+            io.vertx.mutiny.redis.client.Request.cmd(ZUNION),
+            (request, s) -> request.arg(s),
+            (request, request2) -> request);
+
+    return await(mutinyRedis.send(requestWithArgs));
+}
+----
+
+- Updating the https://github.com/quarkusio/quarkus/blob/main/extensions/redis-client/runtime/src/main/java/io/quarkus/redis/client/reactive/ReactiveRedisClient.java[`ReactiveRedisClient`] interface, i.e.:
+
+[source,java]
+----
+Uni<Response> zunion(List<String> args);
+
+Response zunionAndAwait(List<String> args);
+----
+
+- Updating the https://github.com/quarkusio/quarkus/blob/main/extensions/redis-client/runtime/src/main/java/io/quarkus/redis/client/runtime/ReactiveRedisClientImpl.java[`ReactiveRedisClientImpl`], i.e.:
+
```

Comment:
This is really good reference content. But it covers adding the missing commands on the extension which is good but the downside to the approach is that it might require a new release of the extension (Quarkus release) before the user can access the command.

Can we add a section on how the user can add the code new command on their own application? The idea will be the same but this time injecting the `MutinyRedis` client into their application code.

Reply:
Yes, I see.

The description was initially meant to describe how it's possible to add a specific command to `quarkus-redis-client` without waiting for the changes in `vertx-redis-client` - which of course requires a release - as you commented.

I can further describe how this can be done without changes in the `quarkus-redis-client` extension, so this will be more "developer friendly".